### PR TITLE
Delete internal API docs after generation

### PR DESCRIPTION
**Pull request by sheidkamp**
### Description
Delete external API docs after generation

### Docs changes
Delete the generated internal API docs after generation

~~May require a manual update of existing files in firebase~~ Confirmed with docs that it does not.

## Testing steps
Run `make go-generate-apis fmt`

validate that the removed files are no longer present on the file-system:

```
ls -l docs/content/reference/api/github.com/solo-io/gloo/projects/gloo/api/external/
ls: docs/content/reference/api/github.com/solo-io/gloo/projects/gloo/api/external/: No such file or directory
```

Notes for reviewers
There isn't a good way to validate this e2e/in CI without running against production, so we will need to run this through ci after review merge, ideally with https://github.com/k8sgateway/k8sgateway/pull/10334 which needs the same validation.

## Checklist:
 I have performed a self-review of my own code
 I have commented my code, particularly in hard-to-understand areas
 I have made corresponding changes to the documentation
 I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by Rachael-Graham — CHANGES_REQUESTED** [review]
Reposting from Slack for visibility - 

In several of our gloo api docs we link to a bunch of these pages, so many links will be broken if we delete these.
I had a hunch that we autogenerate some of these links. for example, in the [settings.proto reference docs](https://github.com/solo-io/gloo/blob/main/docs/content/reference/api/github.com/solo-io/gloo/projects/gloo/api/v1/settings.proto.sk.md?plain=1#L569), we link to the consul query_options proto (which is one of the files being deleted). but in the actual [settings proto](https://github.com/solo-io/gloo/blob/main/projects/gloo/api/v1/settings.proto#L415) we don’t hardcode a link.

in solo-kit it looks like the [API ref doc template uses a linkForField](https://github.com/solo-io/solo-kit/blob/main/pkg/code-generator/docgen/templates/markdown/proto_file_template.go#L69C1-L69C226) function. the code for that function is [here](https://github.com/solo-io/solo-kit/blob/main/pkg/code-generator/docgen/funcs/template_funcs.go#L278).

long story short - i think we would need to change that linkForField function in solo-kit to somehow recognize when the field is one of these external protos that we will no longer be creating docs for, and either simply do not create a link for it, or hard-code a link to the appropriate external doc (like a link to envoy api docs etc)

**Review by Rachael-Graham — APPROVED** [review]
Thank you!!
